dataclasses_avroschema/fields/fields.py

```diff
@@ -692,7 +692,13 @@ def get_avro_type(self) -> typing.Union[str, typing.List, typing.Dict]:
         return record_type
 
     def default_to_avro(self, value: "schema_generator.AvroModel") -> typing.Dict:
-        return value.to_dict()
+        result = {}
```

Comment:
Here you can use the property `value.schema_def.fields_map` to achieve the same result:

```python
return {
    field_name: field.default_to_avro(getattr(value, field_name))
    for field_name, field in value.schema_def.fields_map.items()
}
```

PS: mypy will complaint, so you can add a new method in the base class to expose the map, similar to this https://github.com/marcosschroh/dataclasses-avroschema/blob/master/dataclasses_avroschema/schema_generator.py#L99. Mayve it can be called `get_fields_map`

Reply:
Oh nice thanks for the hint 😊 This is quite smart.